### PR TITLE
Support k8s secret/configmaps as smallrye config locations

### LINKED ISSUE #14525: configMap configured in quarkus.openshift properties does not get picked up by Quarkus application
**Describe the bug**
`configMap` configured in  `quarkus.openshift` according to [Quarkus - Deploying on OpenShift](https://quarkus.io/guides/deploying-to-openshift) does not get properly recognised by a Quarkus application. This approach worked in 1.7, therefore this is a breaking change.

With Quarkus 1.7, using the properties from docs was sufficient for a RESTful Quarkus application to pick up `configMap`s:

```
quarkus.openshift.config-map-volumes.app-config.config-map-name=app-config
quarkus.openshift.mounts.app-config.path=/deployments/config
```

It seems that with Quarkus 1.11, a new property is required:

```
quarkus.openshift.env.vars.smallrye-config-locations=/deployments/config
```

**Expected behavior**
`configMap` is picked up by application deployed to OpenShift.

**Actual behavior**
`configMap` is not picked up by application deployed to OpenShift.

**To Reproduce**
1. Clone https://github.com/quarkus-qe/quarkus-openshift-test-suite
2. Run `configMap` tests to reproduce the issue:
   ```
   mvn -B -V clean verify -pl app-metadata/deployment,app-metadata/runtime,common,configmap/file-system
   ```
3. Add `quarkus.openshift.env.vars.smallrye-config-locations=/deployments/config` to `config-secret/file-system/src/main/resources/application.properties`. Rerun test for workaround.

**Configuration**
```properties
quarkus.openshift.expose=true
quarkus.openshift.config-map-volumes.app-config.config-map-name=app-config
quarkus.openshift.mounts.app-config.path=/deployments/config
quarkus.s2i.base-jvm-image=registry.access.redhat.com/ubi8/openjdk-11:latest
```

**Screenshots**
N/A

**Environment (please complete the following information):**
 - Output of `uname -a` or `ver`:  `Linux tigris 5.10.8-200.fc33.x86_64 #1 SMP Sun Jan 17 19:53:01 UTC 2021 x86_64 x86_64 x86_64 GNU/Linux`
 - Output of `java -version`: 
    ```
   openjdk version "11.0.9.1" 2020-11-04
   OpenJDK Runtime Environment 18.9 (build 11.0.9.1+11)
   OpenJDK 64-Bit Server VM 18.9 (build 11.0.9.1+11, mixed mode, sharing)
   ```
 - GraalVM version (if different from Java): N/A
 - Quarkus version or git rev: `1.11.0.Final`
 - Build tool (ie. output of `mvnw --version` or `gradlew --version`): `Apache Maven 3.6.1 (d66c9c0b3152b2e69ee9bac180bb8fcc8e6af555; 2019-04-04T21:00:29+02:00)`
 - OpenShift 4.6.9

**Additional context**
(Add any other context about the problem here.)

### PR DESCRIPTION
**Pull request by iocanel**
Resolves: #14525

### CODE REVIEW

File: docs/src/main/asciidoc/deploying-to-kubernetes.adoc [hunk 1]
```diff
@@ -448,6 +448,34 @@ quarkus.kubernetes.secret-volumes.my-volume.secret-name=my-secret
 quarkus.kubernetes.config-map-volumes.my-volume.config-map-name=my-secret
 ----
 
+==== Passing application configuration
+
+Smallrye supports passing configuration from extenral locations. This usually requires additional environment variables or system properties.
```
Comment (geoand) [inline docs/src/main/asciidoc/deploying-to-kubernetes.adoc:453]:
```suggestion
Quarkus supports passing configuration from external locations (via Smallrye Config). This usually requires setting an additional environment variable or system propertiy.
```

File: docs/src/main/asciidoc/deploying-to-kubernetes.adoc [hunk 2]
```diff
@@ -448,6 +448,34 @@ quarkus.kubernetes.secret-volumes.my-volume.secret-name=my-secret
...
+----
+
+or
+
+[source,properties]
+----
+quarkus.kubernetes.app-config-map=<name of the config map containing the configuration>
+----
+
+When these properties are used, the generated manfiests will contain everything required.
```
Comment (geoand) [inline docs/src/main/asciidoc/deploying-to-kubernetes.adoc:474]:
```suggestion
When these properties are used, the generated manifests will contain everything required.
```

**Review by geoand — COMMENTED** [review]
I added some suggestions about the doc update.

Otherwise, LGTM

**Review by geoand — APPROVED** [review]

**Comment by gsmet** [comment]
Is there a specific reason why you added the backport label? It looks like a new feature to me?

**Comment by geoand** [comment]
Yes, I added it because without it there is  large degradation in the user experience of openshift extension (see linked issue)

**Comment by gsmet** [comment]
OK, makes sense, thanks!

**Comment by geoand** [comment]
NP!

**Pull request merged**